src/ConcordiumNetSdk/Types/Memo.cs

```diff
@@ -1,28 +1,56 @@
 using System.Formats.Cbor;
 
+using ConcordiumNetSdk.Helpers;
+using ConcordiumNetSdk.Transactions;
+
 namespace ConcordiumNetSdk.Types;
 
 /// <summary>
-/// A Memo is stored on chain as just an array of bytes.
-/// Convention is to encode a text message as CBOR, but this is not enforced by nodes.  
+/// Represents a memo to be registered on-chain with the <see cref="TransferWithMemo"/> account transaction.
+///
+/// The memo can be any data which is at most <see cref="MaxLength"/> bytes, but convention is to encode a text message as CBOR.
 /// </summary>
-public class Memo
+public readonly struct Memo : IEquatable<Memo>
```

Comment:
Memo is the same as "Data". I recommend removing Data.

Reply:
So in the original SDK CBOR functionality for encoding a string as CBOR was present for the `Memo` class. Do we want to provide this functionality, and if so, does this make sense for both memos and registered data?

I propose making `Memo` into an abstract record, `OnChainData`, and inherit `DataToRegister` (renamed `Data`) and `Memo` for use with `TransferWithMemo` and `RegisterData` transactions, respectively?

Reply:
Yes, it makes sense to use CBOR for both register data and memos. That is what people should do in fact.

Reply:
Okay, makes sense 👍🏼